### PR TITLE
Upstream merge 1.0.3

### PR DESCRIPTION
**Pull request by elenm**
Merge changes from the upstream bloom repository.

I'll create a separate issue to remove California specific information that was added with this merge.

### CODE REVIEW

**Comment by willrlin** [comment]
Adding Andy and Patrick for conflicts in backend/core.

**Comment by elenm** [comment]
> Adding Andy and Patrick for conflicts in backend/core.

Yeah I just deleted any new upstream changes for now so that it doesn't affect Andy's and Patrick's work, but feel free to add them back if they make sense after the merge!

**Review by anders-schneider — APPROVED** [review]
I took a look at all of the changes to files in `backend/core` and they all LGTM

(I didn't look at any other files, though)

**Comment by elenm** [comment]
> I took a look at all of the changes to files in `backend/core` and they all LGTM
> 
> (I didn't look at any other files, though)

There were conflicts in listings-importer.ts, so it might be worth reviewing the [upstream file](https://github.com/bloom-housing/bloom/blob/master/backend/core/scripts/listings-importer.ts) to see if there is anything you want to pull down from there.

**Review by pfeeny5 — APPROVED** [review]
Looked at backend/core and see no conflicts with our updates, mostly just new JSON entries.  LGTM

I did notice they also switched from node version 14 back to 12, so that's worth calling out here.

File: backend/core/package.json [hunk 1]
```diff
@@ -112,7 +112,7 @@
     "testEnvironment": "node"
   },
   "engines": {
-    "node": "14",
+    "node": "12",
```
Comment (avaleske) [inline backend/core/package.json:115]: Can we do this change as a separate PR so it's clear what came from upstream?

File: CHANGELOG.md [hunk 2]
```diff
@@ -44,18 +50,23 @@ All notable changes to this project will be documented in this file. The format
   - Fix repetition of select text on preferences ([#1270](https://github.com/bloom-housing/bloom/pull/1270)) (Emily Jablonski)
   - Fix aplication submission and broken test ([#1270](https://github.com/bloom-housing/bloom/pull/1282)) (Dominik Barcikowski)
   - Fix broken application search in Partners ([#1301](https://github.com/bloom-housing/bloom/pull/1301)) (Dominik Barcikowski)
+    <<<<<<< HEAD
```
Comment (avaleske) [inline CHANGELOG.md:53]: Guessing this was one we missed?

**Review by avaleske — COMMENTED** [review]
Can we name this with the version of upstream we're merging? (I think it's 1.0.3?)

File: CHANGELOG.md (hunk 2 above)
Comment (elenm) [reply CHANGELOG.md:53]: Oh I think this [merge conflict omission](https://github.com/bloom-housing/bloom/blob/master/CHANGELOG.md) is part of their upstream changes. I can fix it here though.

File: backend/core/package.json (hunk 1 above)
Comment (elenm) [reply backend/core/package.json]: I think upstream is also [version 12](https://github.com/bloom-housing/bloom/blob/master/backend/core/package.json).  But I can change this back and do a separate PR for it if you want.

If we want to be super strict about it, I can remove the style change that I added to fix the logo styling that was affected when I pulled down the upstream changes.

This is what it looks like after the merge without my additional style changes:
image

I removed my style changes for now.

Comment (elenm) [reply backend/core/package.json]: Also if we want to be strict about not making any additional changes, I should probably revert my fix in CHANGELOG.md?

**Comment by avaleske** [comment]
Sorry, I hit submit before I meant to earlier.

Thinking about it more, I think you were right and it's ok to make changes in these merge commits to fix any merge conflicts or styling issues. An upside of that is that the changes we make to their code are captured in the same place as the upstream merge.

The reason I suggested having the node engine change in its own PR is cause it wasn't related to fixing the merge, and I didn't want it to look like we downgraded the upstream node version.

Do make sure we don't squash+merge for this. It should be a Merge Commit (which isn't enabled for this branch. @willrlin is that a setting you need to flip?)

**Review by avaleske — APPROVED** [review]

**Comment by willrlin** [comment]
I changed the setting to enable merge commits for main. I agree that it's fine to make changes in the merge commit that fix things that break when merging upstream.

File: sites/public/pages/applications/financial/income.tsx [hunk 3]
```diff
@@ -22,32 +21,33 @@ import FormBackLink from "../../../src/forms/applications/FormBackLink"
...
-  const compareMax = period === "perMonth" ? monthlyMax : annualMax
-
-  if (income < compareMin) {
-    return "low"
-  } else if (income > compareMax) {
-    return "high"
-  }
-  return null
-}
+// type IncomePeriod = "perMonth" | "perYear"
```
[the remark this answers is not in the document]
Comment (willrlin) [reply sites/public/pages/applications/financial/income.tsx:24]: Looks commented out in their master branch: https://github.com/bloom-housing/bloom/blob/master/sites/public/pages/applications/financial/income.tsx

**Review by willrlin — APPROVED** [review]

**Comment by elenm** [comment]
Okay great -- thanks everyone! 
Going to keep the styling changes separate for now, since I've already moved them. I will keep in mind for the future that I can include fixes in the merge PR.
